src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    protected void getStatistic(String fromFileName, String toFileName) {
+        String[] fileInfo = this.readFromFile(fromFileName);
+        int supplySum = 0;
+        int buySum = 0;
+        for (String s : fileInfo) {
+            String[] stringInfo = s.split(",");
+            if (stringInfo[0].equals("supply")) {
+                supplySum += Integer.parseInt(stringInfo[1]);
+            } else {
+                buySum += Integer.parseInt(stringInfo[1]);
+            }
+        }
+        this.writeToFile(toFileName, supplySum, buySum);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
```

Comment:
Please use try with resources ([common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#stick-to-method-naming-convention))